modules/docs/arrow-docs/docs/docs/typeclasses/foldable/README.md

```diff
@@ -9,8 +9,520 @@ permalink: /docs/typeclasses/foldable/
+```
+
+```kotlin:ank
+getLengthFromLeft(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromLeft(None, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromLeft(strList, ListK.foldable())
+```
+
+### ReduceRightOption
+Reduce the elements of this structure down to a single value by applying the provided aggregation function in
+a right-associative manner.
+
+Return None if the structure is empty, otherwise the result of combining the cumulative right-associative
+result of the f operation over the A elements.
+
+```kotlin:ank:silent
+fun <F> getLengthFromRight(strKind: Kind<F, String>, FO: Foldable<F>): Option<String> =
+  FO.run {
+    strKind.reduceRightOption { value: String, base: Eval<String> -> base.map { it + value } }
+      .value()
+  }
+```
+
+```kotlin:ank
+getLengthFromRight(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromRight(None, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromRight(strList, ListK.foldable())
+```
+
+### FoldMap
+Fold implemented by mapping A values into B and then combining them using the given `Monoid<B>` instance.
+
+```kotlin:ank:silent
+fun <F> getLenght(strKind: Kind<F, String>, FO: Foldable<F>): Int =
+  FO.run {
+    strKind.foldMap(Int.monoid()) { it.length }
+  }
+```
+
+```kotlin:ank
+getLenght(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+getLenght(None, Option.foldable())
+```
+
+```kotlin:ank
+getLenght(strList, ListK.foldable())
+```
+  
+### Traverse 
+A typed values will be mapped into `G<B>` by function `f` and combined using `Applicative#map2`.
+
+This method is primarily useful when `G<_>` represents an action or effect, and the specific `A` aspect of `G<A>` is
+not otherwise needed.
+
+```kotlin:ank:silent
+import arrow.instances.either.applicative.applicative
+
+fun <F> traverse(strKind: Kind<F, String>, FO: Foldable<F>): Either<Int,Unit> =
+  FO.run {
+    strKind.traverse_(Either.applicative<Int>()) { Right(it.length) }
+  }.fix()
+```
+
+```kotlin:ank
+traverse(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+traverse(None, Option.foldable())
+```
+
+```kotlin:ank
+traverse(strList, ListK.foldable())
+```
+
+### Sequence
+Similar to traverse except it operates on `F<G<A>>` values, so no additional functions are needed.
+
+```kotlin:ank:silent
+import arrow.instances.option.applicative.applicative
+
+fun <F> sequence(strKind: Kind<F, Kind<ForOption, String>>, FO: Foldable<F>):Option<Unit> =
+  FO.run {
+    strKind.sequence_(Option.applicative())
+  }.fix()
+  
+val maybeStrOpt = Some("abc".some())
+val strNoneList = listOf("a".some(), None, "c".some()).k()
+val strOptList = listOf("a".some(), "b".some(), "c".some()).k()
+```
+
+```kotlin:ank
+sequence(maybeStrOpt, Option.foldable())
+```
+
+```kotlin:ank
+sequence(None, Option.foldable())
+```
+
+```kotlin:ank
+sequence(strNoneList, ListK.foldable())
+```
+
+```kotlin:ank
+sequence(strOptList, ListK.foldable())
+```
+
+### Find
+Find the first element matching the predicate, if one exists.
```

Comment:
`If there are no elements, the result is None.`

For consistency with the ones below :D